zerver/views/webhooks/gosquared.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+from zerver.models import Client, UserProfile
+
+from django.http import HttpRequest, HttpResponse
+from six import text_type
+from typing import Dict, Any, Optional
+
+BODY_TEMPLATE = '{website_name} has {user_num} visitors online.'
+
+@api_key_only_webhook_view('GoSquared')
+@has_request_variables
+def api_gosquared_webhook(request, user_profile, client,
+                          payload=REQ(argument_type='body'),
+                          stream=REQ(default='gosquared'),
+                          topic=REQ(default=None)):
+    # type: (HttpRequest, UserProfile, Client, Dict[str, Dict[str, Any]], text_type, text_type) -> HttpResponse
+    try:
+        domain_name = payload['siteDetails']['domain']
+        user_num = payload['concurrents']
+        body = BODY_TEMPLATE.format(website_name=domain_name, user_num=user_num)
```

Comment:
Please pull this line out of the try/except (`body = BODY_TEMPLATE...`), as if it gets a KeyError, it's a programming error, not a missing key in the JSON.  In general, try to limit how many lines of code you put into a try/except block.